File: MinJu/231211_week/Programmers_12909_올바른 괄호.py
def solution(s):
    answer = True
    arr = []   
  
    for i in s:
        if i == '(':  
            arr.append(i) 
        else: 
            if arr == []: 
                answer = False
                break
            else:
                arr.pop() 

    if arr != []: 
        answer = False

    return answer

File: SooYeon/240304_week/Programmers_12909_올바른 괄호.js
function solution(s){
  let answer = true;

  let stack = [];
  for(let i = 0; i < s.length; i++) {
      if(s[i] === '(') {
          stack.push(s[i]);
      } 
      else {
          if(stack[stack.length - 1] === '(') {
              stack.pop();
          } else {
              answer = false;
              break;
          }
      }
  }
  
  if(stack.length) answer = false;

  return answer;
}